### PR TITLE
Determine iptables backend in container add-snat-rule-to-upstream-dns.

### PR DESCRIPTION
**Pull request by axel7born**
The calico-node container determines the backend of iptables in the calico-node executable based on existing chains in the mangle table. The container `add-snat-rule-to-upstream-dns` adds additional rules to masquerade traffic to the upstream DNS server. These rules must added with the same iptables version as used by calico-node. For that reason this PR adds a iptables-backend detection to this container.

In addition to that, we mount the xtables lock file into `add-snat-rule-to-upstream-dns`. 

**How to categorize this PR?**

/area networking
/kind enhancement

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
Determine iptables backend in container add-snat-rule-to-upstream-dns.
```

### CODE REVIEW

File: charts/internal/calico/templates/node/daemonset-calico-node.yaml [hunk 1]
```diff
@@ -173,10 +173,29 @@ spec:
...
+            - mountPath: /run/xtables.lock
+              name: xtables-lock
+              readOnly: false
           command:
           - /bin/sh
           - -c
-          - "while true; do sleep 15; for i in $(cat /etc/resolv.conf | grep nameserver | sed -n -e 's/^.*nameserver //p' ); do iptables -t nat -C POSTROUTING -s ${POD_CIDR} -d $i/32 ! -o cali+ -m comment --comment \"calico masquerade non-cluster\" -j MASQUERADE 2>/dev/null || iptables -t nat -I POSTROUTING 1 -s ${POD_CIDR} -d $i/32 ! -o cali+ -m comment --comment \"calico masquerade non-cluster\" -j MASQUERADE; done; sleep 45; done"
+          - |
+            nft_kubelet_rules=$( (iptables-nft-save -t mangle || true; ip6tables-nft-save -t mangle || true) 2>/dev/null | grep -E '^:(KUBE-IPTABLES-HINT|KUBE-KUBELET-CANARY)');
+            if [[ $nft_kubelet_rules != "" ]]; then
```
Comment (DockToFuture) [inline charts/internal/calico/templates/node/daemonset-calico-node.yaml:185]:
```suggestion
            if [[ -n $nft_kubelet_rules ]]; then
```

File: charts/internal/calico/templates/node/daemonset-calico-node.yaml [hunk 2]
```diff
@@ -173,10 +173,29 @@ spec:
             requests:
               cpu: 10m
               memory: 50Mi
+          volumeMounts:
+            - mountPath: /run/xtables.lock
+              name: xtables-lock
+              readOnly: false
```
Comment (ScheererJ) [inline charts/internal/calico/templates/node/daemonset-calico-node.yaml:179]: Nit: Usually the `volumeMounts` are at the end of the container specification. Not really a problem, but I was just a bit surprised.

**Review by ScheererJ — CHANGES_REQUESTED** [review]
Thanks for adding proper nft/legacy support.

**Review by DockToFuture — APPROVED** [review]
/lgtm :)

**Review by ScheererJ — APPROVED** [review]
/lgtm

**Pull request merged**